extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java

```diff
@@ -209,6 +215,94 @@ private void operatingSystemMetrics(MetricRegistry registry, List<String> names)
                 .build();
         registry.register(meta, new LambdaGauge(() -> operatingSystemMXBean.getAvailableProcessors()));
         names.add(CPU_AVAILABLE_PROCESSORS);
+
+        // some metrics are only available in jdk internal class 'com.sun.management.OperatingSystemMXBean': cast to it.
+        // com.sun.management.OperatingSystemMXBean is not available in SubstratVM
+        if (!ImageInfo.inImageCode()) {
+            try {
+                // the cast will fail for some JVM not derived from HotSpot (J9 for example)
+                com.sun.management.OperatingSystemMXBean internalOperatingSystemMXBean = (com.sun.management.OperatingSystemMXBean) operatingSystemMXBean;
+                meta = Metadata.builder()
+                        .withName(PROCESS_CPU_LOAD)
+                        .withType(MetricType.GAUGE)
+                        .withDisplayName("Process CPU load")
+                        .withDescription("Displays  the \"recent cpu usage\" for the Java Virtual Machine process. " +
+                                "This value is a double in the [0.0,1.0] interval. A value of 0.0 means that none of " +
+                                "the CPUs were running threads from the JVM process during the recent period of time " +
+                                "observed, while a value of 1.0 means that all CPUs were actively running threads from " +
+                                "the JVM 100% of the time during the recent period being observed. Threads from the JVM " +
+                                "include the application threads as well as the JVM internal threads. " +
+                                "All values betweens 0.0 and 1.0 are possible depending of the activities going on in " +
+                                "the JVM process and the whole system. " +
+                                "If the Java Virtual Machine recent CPU usage is not available, the method returns a negative value.")
+                        .build();
+                registry.register(meta, new LambdaGauge(() -> internalOperatingSystemMXBean.getProcessCpuLoad()));
```

Comment:
In general we try and avoid lambdas in the runtime code. On hotspot they use considerably more memory than the equivalent anonymous inner class.

Reply:
Lambdas are similarly used in a lot of places in this recorder, do we want to turn all of them into inner classes? I guess yeah

Reply:
`LambdaGauge` is a Quarkus class created for it's metrics implementation. I prefere to keep it to be consistent. I also is more readable.

@stuartwdouglas, I'm curious, can you point me to a link that explain why lambdas uses more memory than anonymous inner class? I know that there is some performance tradeoff (lambda peak performance is achieved later than anonymous inner class) but I'm not aware of a memory usage issue. In fact ready this deep dive article (https://www.infoq.com/articles/Java-8-Lambdas-A-Peek-Under-the-Hood/) non capturing lambdas are translated as static methods and should use less memory than anonymous inner classes that are real classes stored in Metaspace.